src/include/miopen/unsortedsegmentsum/problem_description.hpp

```diff
@@ -0,0 +1,155 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include "miopen/miopen.h"
+#include <miopen/activ.hpp>
+#include <miopen/problem_description_base.hpp>
+#include <miopen/tensor.hpp>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace UnsortedSegmentSum {
+
+struct FwdProblemDescription : ProblemDescriptionBase
+{
+    FwdProblemDescription(const TensorDescriptor& InputDesc_,
+                          const TensorDescriptor& OutputDesc_,
+                          const TensorDescriptor& SegmentIdsDesc_)
+        : InputDesc(InputDesc_), OutputDesc(OutputDesc_), SegmentIdsDesc(SegmentIdsDesc_)
```

Comment:
add conditions based on modnn https://github.com/moreh-dev/backend/blob/95cf0105e304b0b7d7ec065f192a5207f6b3fe5a/modnn/core/op/unsorted_segment_sum.cpp#L32

Reply:
conditions based on modnn are already included, they are the same with IsValidDims()

Reply:
How can you check for the condition `data_tensor.dims_size() >= segment_ids_tensor.dims_size()` at `IsValidDims()` ?

Reply:
this is restriction in document